### PR TITLE
Make StatusListPicker work with backend models

### PR DESCRIPTION
**Pull request by osmaczko**
closes: https://github.com/status-im/status-desktop/issues/6500

### Checklist

- [x] follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
  - the scope should be the component's name e.g: `feat(StatusListItem): ... `
  - when adding new components, the scope is the module e.g: `feat(StatusQ.Controls): ...`
- [ ] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [ ] test changes in both light and dark theme?
- [ ] is this a breaking change?
    - [ ] use the dedicated `BREAKING CHANGE` commit message section
    - [ ] resolve breaking changes in [status-desktop](https://github.com/status-im/status-desktop)
        - [ ] (pre-merge) adapt code to breaking changes
        - [ ] (post-merge) update StatusQ submodule pointer
- [x] test changes in [status-desktop](https://github.com/status-im/status-desktop)

### CODE REVIEW

**Comment by osmaczko** [comment]
@noeliaSD could you please check in the sandbox app if `StatusListPicker` works as expected?

**Review by stefandunca — APPROVED** [review]
Beautifully simple.

File: src/StatusQ/Components/StatusListPicker.qml [hunk 1]
```diff
@@ -197,40 +175,43 @@ Item {
...
-                     // Clear all the list
-                    root.inputList.get(i).selected = false
-                }
+        // If it is a single selection we must ensure that when a new key is selected others are cleared.
+        id: itemsSelectorHelper
+
+        function selectItem(modelData, checked) {
+            modelData.selected = checked
+            if (!root.multiSelection && checked) {
+                itemsSelectorHelper.uncheckOthersThan(modelData.key)
```
Comment (caybro) [inline src/StatusQ/Components/StatusListPicker.qml:199]: I wonder why is this needed for? If you put all the checkboxes/radio buttons in a `ButtonGroup` (https://doc.qt.io/qt-5/qml-qtquick-controls2-buttongroup.html) then it should be done automatically

**Review by caybro — COMMENTED** [review]
Quite an improvement! But I still think it could be done even simpler :)

**Review by noeliaSD — CHANGES_REQUESTED** [review]
Nice to see the sort proxy model integrated here!! :)

Tested and some issues found in sandbox:

1. Initialization in `Multiple selection` and `Multiple selectin and diplayed names...` is not correct. Shown USD and GBP but selected EUR and ETH in both cases.
2. Multiple selection option doesn't update the displayed name.
3. Single selection: New selection not updated when the searcher is used.

Attached a video for better understanding: 

https://user-images.githubusercontent.com/97019400/180936558-a98d009b-17da-466b-a83d-2be80152a140.mov

**Comment by osmaczko** [comment]
@noeliaSD great findings (seems I didn't perform tests with care :sweat: ). All of them should be fixed now, could you please re-check?

File: src/StatusQ/Components/StatusListPicker.qml (hunk 1 above)
Comment (osmaczko) [reply src/StatusQ/Components/StatusListPicker.qml]: Yeah, that's a good question. Unfortunately, the delegate of `StatusListPicker` is not `AbstractButton` based..

**Review by noeliaSD — APPROVED** [review]
LGTM!
